wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenConnectionPlugin.java

```diff
@@ -171,7 +171,10 @@ public Connection connect(
 
         if (conn == null) {
 
-          this.bgStatus = this.pluginService.getStatus(BlueGreenStatus.class, this.bgdId);
+          BlueGreenStatus latestStatus = this.pluginService.getStatus(BlueGreenStatus.class, this.bgdId);
```

Comment:
That increases a chance to use a stale BG status. No BG status means that BGD doesn't exist so no routing should be applied.